### PR TITLE
Build six complete CMS templates with native editing and recovery

### PR DESCRIPTION
**Pull request by punk6529**
## Issue

The CMS template catalogue did not give people complete, distinctive examples they could quickly turn into their own websites. Editing galleries and richer page content also needed clearer controls.

## Fix

Replace the starter catalogue with six fully populated designs—Personal, Artist, Collector, Meme, Organization, and Fund / DAO—and render them as editable native CMS documents. Existing documents continue using their original renderer.

## Changes

- Add 71 example pages with artwork archives, individual work pages, programme listings, project mockups, contact forms, and collection records. Include credited art assets and licensed local fonts.
- Add separate Preview and Use template buttons, page selection, direct section selection, and structured gallery, row, mockup, and poster controls. Preserve design customization, undo, drafts, and advanced tools.
- Preserve canonical artwork order and annotations through edits, validation, and static recovery. Validate optional native presentation fields without changing the backend contract.
- Update translated controls, builder/recovery documentation, and the Help Bot index.

## Validation

- Production build passed; strict changed-file lint and TypeScript checks passed.
- All 635 CMS tests passed across 56 suites, including legacy rendering, gallery ordering, editing, package round trips, and recovery. The full Jest type ratchet passed with no baseline increase.
- Visually reviewed all six designs and checked all 71 pages at desktop and 390px mobile widths. Exercised template selection, navigation, search/filter/list views, editing, undo, and design controls.
- All six final packages passed live backend validation with no issues and matching hashes. Validation did not save drafts or publish sample content.
- Authenticated publication was not performed under a real profile during preview QA; existing publication tests and non-persisting backend validation cover the contract.

## Risk

- Level: Medium
- Why: New rendering and editing paths affect newly selected templates. Legacy documents retain their existing renderer, and no database or backend deployment is required.
- Rollback: Revert the frontend change through the normal deployment workflow. Native content remains recoverable, though reverting the new renderer removes its specialized presentation.

## Review Notes

Focus on gallery ordering and annotation preservation, internal page links, optional native field validation, and recovery of multi-page documents. Sample identities and claims are fictional and require owner review before publication.

### CODE REVIEW

**Comment by punk6529** [comment]
# PR #3984 review dispositions

Original review evidence below is pinned to head `b988b8c620c6b1a19de938d925fe193b234721d7`. The focused follow-up is `55344f9f8361f2cbf361564b4a3e79f3ff92eb27`; source links remain pinned to the reviewed original where the disposition describes unchanged behavior.

## Review coverage

CodeRabbit reported a success status but **did not perform a review**: its [comment](https://github.com/6529-Collections/6529seize-frontend/pull/3984#issuecomment-5648710038) explicitly reports the review limit. The Claude general and i18n reviews state that 48 implementation files were omitted. Their concerns were checked against the actual source; neither comment is evidence of a complete implementation review.

## Claude general

[Original comment](https://github.com/6529-Collections/6529seize-frontend/pull/3984#issuecomment-5648736011)

- **“Legacy template constant renamed with no backward-compatibility path.”** The chooser intentionally offers six approved templates, but the original 39 definitions remain in `LEGACY_STUDIO_TEMPLATES`, and explicit instantiation searches both registries ([templates.ts:26–62](https://github.com/6529-Collections/6529seize-frontend/blob/b988b8c620c6b1a19de938d925fe193b234721d7/lib/profile-cms/studio/templates.ts#L26)). Existing saved records contain the full package, which the loader validates directly; they are not reconstructed from a chooser ID ([package-normalize.ts:68](https://github.com/6529-Collections/6529seize-frontend/blob/b988b8c620c6b1a19de938d925fe193b234721d7/lib/profile-cms/builder/package-normalize.ts#L68)). The approved renderer is opt-in, with legacy rendering retained ([CmsStudioSiteRenderer.tsx:95](https://github.com/6529-Collections/6529seize-frontend/blob/b988b8c620c6b1a19de938d925fe193b234721d7/components/profile-cms/CmsStudioSiteRenderer.tsx#L95)). Existing tests instantiate `signature` and verify it stays on the legacy renderer, including unknown-design fallback ([test:44](https://github.com/6529-Collections/6529seize-frontend/blob/b988b8c620c6b1a19de938d925fe193b234721d7/__tests__/components/profile-cms/CmsApprovedSiteRenderer.test.tsx#L44)). No migration or publication rewrite occurs. A further legacy recovery test would expand coverage; no broken loading path was identified.

- **Visitor correspondence storage boundary.** Drafts live in a bounded browser-module memory map, keyed by profile handle and package, with a maximum of 24 scopes and 24 form drafts per scope. The server never remembers them; there is no localStorage, sessionStorage, cookie or API persistence ([session.tsx:19–60](https://github.com/6529-Collections/6529seize-frontend/blob/b988b8c620c6b1a19de938d925fe193b234721d7/components/profile-cms/approved-renderer/session.tsx#L19)). Reloading or closing the page clears this application memory. The form intentionally retains unsent text during navigation in the same open page. It does not infer a new anonymous visitor identity: another person using that same open tab could see an unfinished draft, as with an unsent form. Cross-profile isolation is covered; this is not a claim of kiosk-session isolation. Preparing email opens an explicit mail client link and does not send a message.

- **Header-injection tests for body/other mailto parameters.** Static recovery emits recipient and an optional subject, with recipient validation and CR/LF stripping on the bounded subject; it does not emit a body ([static-site.ts:432–439](https://github.com/6529-Collections/6529seize-frontend/blob/b988b8c620c6b1a19de938d925fe193b234721d7/lib/profile-cms/recovery/static-site.ts#L432)). The interactive form encodes the entire recipient, subject and body separately with `encodeURIComponent` ([Contact.tsx:60–76](https://github.com/6529-Collections/6529seize-frontend/blob/b988b8c620c6b1a19de938d925fe193b234721d7/components/profile-cms/approved-renderer/Contact.tsx#L60)); body line breaks are intentional message formatting, not additional query parameters. No uncovered static body parameter exists.

## Claude i18n

[Original comment](https://github.com/6529-Collections/6529seize-frontend/pull/3984#issuecomment-5648741959)

- **“New approved-renderer UI ships extensive hardcoded English copy.”** The cited strings are English-locale test expectations. Actual search, empty state, clear and view controls use `t(context.locale, ...)` ([Gallery.tsx:118–173](https://github.com/6529-Collections/6529seize-frontend/blob/b988b8c620c6b1a19de938d925fe193b234721d7/components/profile-cms/approved-renderer/Gallery.tsx#L118)); contact labels, hints and actions do likewise ([Contact.tsx:90–150](https://github.com/6529-Collections/6529seize-frontend/blob/b988b8c620c6b1a19de938d925fe193b234721d7/components/profile-cms/approved-renderer/Contact.tsx#L90)). The keys are included in the existing `profileCmsStudio` dictionaries for all five supported locales, rather than a new renderer-specific dictionary ([English keys](https://github.com/6529-Collections/6529seize-frontend/blob/b988b8c620c6b1a19de938d925fe193b234721d7/i18n/messages/profileCmsStudio.en-US.json#L153), [French keys](https://github.com/6529-Collections/6529seize-frontend/blob/b988b8c620c6b1a19de938d925fe193b234721d7/i18n/messages/profileCmsStudio.fr-FR.json#L153)). No duplicate extraction is needed.

- **Counts/pluralization.** Gallery count uses `formatNumber(context.locale, visible.length)` and separate translated singular/plural messages for the five supported languages ([Gallery.tsx:130–139](https://github.com/6529-Collections/6529seize-frontend/blob/b988b8c620c6b1a19de938d925fe193b234721d7/components/profile-cms/approved-renderer/Gallery.tsx#L130)). There is no raw English noun concatenation. This is scoped to the current supported locale set.

- **Recovery alt text.** The flagged image element uses escaped authored `asset.alt_text`, or an empty alt, at [static-site.ts:483](https://github.com/6529-Collections/6529seize-frontend/blob/b988b8c620c6b1a19de938d925fe193b234721d7/lib/profile-cms/recovery/static-site.ts#L483). It is not fixed English interface text and must not be automatically translated.

- **Archive-chrome documentation follow-up addressed.** Independent static recovery introduces the fixed link label “6529 profile” ([static-site.ts:254](https://github.com/6529-Collections/6529seize-frontend/blob/b988b8c620c6b1a19de938d925fe193b234721d7/lib/profile-cms/recovery/static-site.ts#L254)). The follow-up adds an [“Independent recovery archive” section](https://github.com/6529-Collections/6529seize-frontend/blob/55344f9f8361f2cbf361564b4a3e79f3ff92eb27/ops/workstreams/i18n-fallbacks/profile-cms-studio.md#L28), naming the exact English surface, user impact, owner and document-language localization follow-up in PR #3984. This is a documented fallback, not a claim that archive chrome is translated.

- **Authored example content.** Artwork titles, credits, fictional prose and editable inquiry subjects remain document content. The existing [sample-language coverage note](https://github.com/6529-Collections/6529seize-frontend/blob/b988b8c620c6b1a19de938d925fe193b234721d7/ops/workstreams/i18n-fallbacks/profile-cms-studio.md) describes that boundary. Changing interface locale does not rewrite or rehash a saved document.

## Additional review dispositions

- **Inspector complexity:** the focused follow-up extracts the row-fallback synchronization helper and replaces the existence predicate, with 33 focused inspector tests passing. The larger mixed legacy/native inspector function remains a maintainability concern; no concrete state or data-integrity defect was established by the complexity score itself. A broader decomposition is deferred to avoid changing working form-state/reset boundaries in this release. This disposition does not treat complexity as harmless indefinitely; the follow-up should preserve existing structured-field and canonical-gallery regression coverage.
- **Sonar S6479, collection-editor row keys:** positional keys at `StudioCollectionFields.tsx:165` do retain native disclosure-open state and focus at a list position when rows move or are removed. Inputs are controlled, and the canonical-array preservation tests cover the actual edits; no hidden content loss was identified. This is deferred authoring polish. Row labels and asset IDs are editable/nonunique, so using them as keys would introduce remount or collision problems. A proper ephemeral-row identity scheme needs explicit move/add/remove/reset handling without adding identities to canonical document content.
- **GLM missing `useSearchParams` mock:** the gallery-auth test intentionally mocks `CmsSiteRenderer` to `null` at lines 27–30, so the descendant contact component containing that hook never mounts. The real-renderer tests supply the hook mock. The alleged render failure is not present in the passing gallery-auth suite.
- **Responsiveness automation:** [run 34719443814](https://github.com/6529-Collections/6529reviewbot/actions/runs/34719443814) did not execute browser checks. Dependency setup called the disabled `6529 install:frozen` command; the exact error says `6529 ci` is the canonical frozen install. This is a reviewer-runner command mismatch, not CMS responsiveness evidence. Use separately recorded actual browser QA; do not present this automation as a pass.
- **Other available reviews:** Claude crypto/security and WCAG comments report no findings while explicitly noting truncated source coverage. The GLM review also notes two unavailable reviewer slices. Preserve those limits in the final review summary.

**Comment by punk6529** [comment]
Review follow-up:

- `55344f9f83` fixes section activation through native Edit buttons, semantic gallery controls/status, and the project mockup's book-cover contrast (now 4.62:1). Mouse editing, Enter/Space activation, nested controls, gallery search/list, and the resulting layout were checked. Sonar passed on the subsequent integrated head.
- `445d3d940f` corrects three test files to construct native block fixtures through the package schema and narrow extension-field reads. The full Jest type ratchet passes without a baseline increase; all 57 affected tests pass. No application behavior changes in this commit.
- GLM's missing-navigation-mock concern does not apply to the gallery-auth test: that suite mocks `CmsSiteRenderer` itself, so the descendant contact component does not mount. The actual suite passes.
- The responsiveness bot failed before browser checks because its workflow invokes the unavailable `install:frozen` command. That run provides no responsive-layout evidence. The manual desktop/mobile review and successful hosted smoke/route-shell runs are reported separately.
- Remaining nonblocking findings: the existing inspector's full component decomposition is deferred; collection row disclosures currently retain open/focus state by position when reordered, while controlled values and canonical artwork ordering remain correct. Introducing stable editor-only row identities warrants a separate change and focused disclosure/focus tests. The static-email regex input is bounded to 254 characters before execution. Archive navigation's English fallback now has an explicit owner and follow-up in the locale coverage note.

Coordinator request [97](https://github.com/6529-Collections/6529-release-coordinator/issues/97) is retained for this release and its production continuation; no duplicate submission is needed.

File: components/profile-cms-builder/studio/ProfileCmsStudioEditor.tsx [hunk 1]
```diff
@@ -281,19 +288,30 @@ export default function ProfileCmsStudioEditor({
...
+            value={pageSearch}
+            onChange={setPageSearch}
+          />
+          <ul className="tw-m-0 tw-max-h-72 tw-list-none tw-space-y-2 tw-overflow-y-auto tw-p-0">
+            {document.payload.pages
+              .filter((item) =>
+                item.metadata.title
+                  .toLocaleLowerCase(locale)
+                  .includes(pageSearch.toLocaleLowerCase(locale))
+              )
```
[the remark this answers is not in the document]
Comment (punk6529) [reply components/profile-cms-builder/studio/ProfileCmsStudioEditor.tsx:305]: Reviewed and addressed in 0b11c77ec6918390335420d5322e95f81d58d898.

Page search now matches both the visible navigation label and the underlying title, using locale-aware case folding. Regression covers the two different fields and no-match behavior; also verified by renaming and finding a page in the browser.

Validation: all 635 CMS tests across 56 suites pass, strict lint and source types pass, and the full Jest type ratchet has no baseline increase.

File: components/profile-cms-builder/studio/StudioBlockInspector.tsx [hunk 2]
```diff
@@ -506,6 +667,35 @@ function isEditableGallery(value: unknown): boolean {
...
+}
+
+function rowsText(rows: readonly Record<string, unknown>[]): string {
+  return rows
+    .map(
+      (row) =>
+        `${getString(row, "label") ?? ""}: ${getString(row, "value") ?? ""}`
+    )
+    .join("\n");
+}
```
[the remark this answers is not in the document]
Comment (punk6529) [reply components/profile-cms-builder/studio/StudioBlockInspector.tsx:699]: Reviewed and addressed in 0b11c77ec6918390335420d5322e95f81d58d898.

Row fallback formatting now omits the colon for an empty label. Regressions cover clearing a label, rendering its value once, and reopening/editing the unlabeled row while preserving independent prose.

Validation: all 635 CMS tests across 56 suites pass, strict lint and source types pass, and the full Jest type ratchet has no baseline increase.

File: components/profile-cms-builder/studio/StudioCollectionFields.tsx [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+                Array.isArray(fields["display_modes"]) &&
+                fields["display_modes"].includes("list")
+              }
+              onChange={(event) =>
+                onChange({
+                  display_modes: event.target.checked
+                    ? ["grid", "list"]
+                    : ["grid"],
+                })
+              }
```
[the remark this answers is not in the document]
Comment (punk6529) [reply components/profile-cms-builder/studio/StudioCollectionFields.tsx]: Reviewed and addressed in 0b11c77ec6918390335420d5322e95f81d58d898.

The toggle now adds/removes only list entries. It preserves imported strings, objects, numeric values, order and duplicates outside that one mode; regressions exercise toggling both directions.

Validation: all 635 CMS tests across 56 suites pass, strict lint and source types pass, and the full Jest type ratchet has no baseline increase.

File: components/profile-cms-builder/studio/StudioTemplateLibrary.tsx [hunk 4]
```diff
@@ -137,13 +129,41 @@ export default function StudioTemplateLibrary({
...
+                  {t(locale, "profileCms.studio.preview")}
+                </StudioButton>
+                <StudioButton
+                  primary
+                  onClick={() =>
+                    onUse(instantiateCmsStudioTemplate(template.id, handle))
+                  }
+                >
+                  {t(locale, "profileCms.studio.useTemplate")}
+                </StudioButton>
```
[the remark this answers is not in the document]
Comment (punk6529) [reply components/profile-cms-builder/studio/StudioTemplateLibrary.tsx:193]: Reviewed and addressed in 0b11c77ec6918390335420d5322e95f81d58d898.

All chooser and preview Use actions now have template-specific accessible names. Added the translated message to all five locale dictionaries and checked all six distinct chooser actions.

Validation: all 635 CMS tests across 56 suites pass, strict lint and source types pass, and the full Jest type ratchet has no baseline increase.

File: components/profile-cms-builder/studio/StudioTemplateLibrary.tsx [hunk 5]
```diff
@@ -152,82 +172,32 @@ export default function StudioTemplateLibrary({
...
-      candidate.id === (imageBlock ? getString(imageBlock, "asset_id") : null)
-  );
-  const source = asset
-    ? (getCmsStudioDemoAssetPath(asset) ??
-      getCmsStudioMemeDisplayAsset(asset)?.localPath)
-    : null;
-  const light = ["paper", "stone"].includes(
-    template.presentation.studio_palette
+  const document = useMemo(
+    () => instantiateCmsStudioTemplate(template.id, handle),
```
[the remark this answers is not in the document]
Comment (punk6529) [reply components/profile-cms-builder/studio/StudioTemplateLibrary.tsx:213]: Reviewed and addressed in 0b11c77ec6918390335420d5322e95f81d58d898.

The library now checks the canonical profile-handle schema before mounting any thumbnail, preview or Use flow. Invalid handles show a translated explanation. Regressions cover invalid inputs, a valid two-character handle and changing an open preview to an invalid handle.

Validation: all 635 CMS tests across 56 suites pass, strict lint and source types pass, and the full Jest type ratchet has no baseline increase.

File: lib/profile-cms/protocol/v1/validation-native-content.ts [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+            typeof assetId === "string" &&
+            (remainingAssets.get(assetId) ?? 0) === 0
+          ) {
+            addIssue(issues, {
+              ...context,
+              path: `${entryPath}/asset_id`,
+              code: "block.gallery_item_unlisted",
+              message:
+                "Each artwork item needs its own placement in the gallery's asset_ids.",
+            });
```
[the remark this answers is not in the document]
Comment (punk6529) [reply lib/profile-cms/protocol/v1/validation-native-content.ts:103]: Reviewed and addressed in 0b11c77ec6918390335420d5322e95f81d58d898.

The proposed validation relaxation is intentionally not applied. For authored native galleries, asset_ids is the canonical placement list and items annotates its occurrences; allowing orphan annotations would defeat the existing lost-annotation guard. The renderer/recovery fallback is defensive compatibility, not publication validity. Missing and empty canonical arrays remain invalid, now explicitly documented and tested. Recovery can nevertheless preserve the artwork safely as addressed in the separate recovery thread.

Validation: all 635 CMS tests across 56 suites pass, strict lint and source types pass, and the full Jest type ratchet has no baseline increase.

File: lib/profile-cms/recovery/static-site.ts [hunk 7]
```diff
@@ -319,7 +376,66 @@ function renderGallery(block: CmsBlockV1, context: RenderContext): string {
...
+        [text(item, "title"), context.assets.get(id)?.alt_text].find(
+          (value) => typeof value === "string" && value.length > 0
+        ) ?? id;
+      const pageId = text(item, "page_id");
+      const titleLink = pageId
+        ? linkToPage(pageId, title, context)
+        : escapeHtml(title);
+      return `<article class="cms-gallery-item">${renderAsset(id, context, undefined, pageId, false)}<h3>${titleLink}</h3>${paragraphs(text(item, "subtitle"))}<small>${escapeHtml(text(item, "category"))}</small></article>`;
+    })
+    .join("");
```
[the remark this answers is not in the document]
Comment (punk6529) [reply lib/profile-cms/recovery/static-site.ts]: Reviewed and addressed in 0b11c77ec6918390335420d5322e95f81d58d898.

Recovery now derives artwork occurrences from items only when both ID arrays are empty. Existing array precedence, occurrence-specific captions, asset lookup, URI validation and escaping are preserved. Regressions cover missing/empty arrays, repeated placements and unsafe URI/markup rejection.

Validation: all 635 CMS tests across 56 suites pass, strict lint and source types pass, and the full Jest type ratchet has no baseline increase.

File: lib/profile-cms/studio/templates-approved-artist.json [hunk 8]
```diff
@@ -0,0 +1,2599 @@
...
+            },
+            {
+              "label": "2023",
+              "value": "2023 Drawing Systems Rowan Press Workshop, London · Open studio and print release"
+            },
+            {
+              "label": "2022",
+              "value": "2022 Plotting Methods Merebank Print Lab, Kent · Research residency"
+            }
+          ],
```
[the remark this answers is not in the document]
Comment (punk6529) [reply lib/profile-cms/studio/templates-approved-artist.json:2067]: Reviewed and addressed in 0b11c77ec6918390335420d5322e95f81d58d898.

Removed duplicated timeline years and separated each project title from its venue with a comma. The archival fallback is regenerated from the corrected rows; focused assertions verify the year occurs only in its label.

Validation: all 635 CMS tests across 56 suites pass, strict lint and source types pass, and the full Jest type ratchet has no baseline increase.

**Comment by punk6529** [comment]
The full CodeRabbit review of `77ceefaaaa75ed3ebda4a6384f0e619198fd661f` was inspected against the implementation.

The gallery placement rule remains intentional: `asset_ids` is the canonical artwork order, and `items` annotates those placements. Missing/empty canonical IDs do not make orphan annotations valid for publishing. The recovery renderer can still preserve artwork defensively from imported item-only documents; rendering tolerance does not grant publication validity.

The thumbnail performance suggestion is deferred. The six instances are memoized per template and handle. A local cold Node measurement of all six package instantiations was 63.61 ms; this is not a mobile browser performance claim. Viewport-based thumbnail mounting remains a useful follow-up, but changes initial-render/hydration behavior and warrants its own browser measurement. No correctness or content-preservation issue depends on that optimization.

The remaining findings are fixed in `0b11c77ec6918390335420d5322e95f81d58d898`, included in current head `20c59aa86cb4557f6f1a99aaa7fdcf93f19d07e6`: page-label search, empty-row fallback formatting, preservation of imported gallery modes, distinguishing accessible template actions, invalid-handle guards, item-only archive fallback, duplicate archive callout suppression, and the Artist/Organization sample text corrections. All 635 CMS tests across 56 suites pass, strict lint and source types pass, and the full Jest type ratchet has no baseline increase. The ten review threads have explicit dispositions and are resolved.

File: lib/profile-cms/studio/templates-approved-organization.json [hunk 9]
```diff
@@ -0,0 +1,599 @@
...
+            },
+            {
+              "label": "Jo Mercer",
+              "value": "Jo Mercer Print studio & workshops"
+            },
+            {
+              "label": "Nia Ellis",
+              "value": "Nia Ellis Visitors & volunteers"
+            }
+          ],
```
[the remark this answers is not in the document]
Comment (punk6529) [reply lib/profile-cms/studio/templates-approved-organization.json:472]: Reviewed and addressed in 0b11c77ec6918390335420d5322e95f81d58d898.

Removed repeated poster titles and team names from the supporting values, kept the row fallback aligned, and removed decorative authored arrows from the affected contact text/title and Visit label. IDs, layout and artwork are unchanged; focused catalogue assertions pass.

Validation: all 635 CMS tests across 56 suites pass, strict lint and source types pass, and the full Jest type ratchet has no baseline increase.

File: lib/profile-cms/studio/templates-approved-personal.json [hunk 10]
```diff
@@ -0,0 +1,686 @@
...
+            },
+            {
+              "label": "Team",
+              "value": "Design lead, with two developers"
+            },
+            {
+              "label": "When",
+              "value": "January–June 2026"
+            }
+          ],
```
[the remark this answers is not in the document]
Comment (punk6529) [reply lib/profile-cms/studio/templates-approved-personal.json:165]: Reviewed and addressed in 0b11c77ec6918390335420d5322e95f81d58d898.

Recovery suppresses prose only when it exactly matches the formatted row fallback. Independent paragraphs remain alongside the record. Regressions cover labeled/unlabeled rows, occurrence counts, independent paragraphs and every approved package's recovered content.

Validation: all 635 CMS tests across 56 suites pass, strict lint and source types pass, and the full Jest type ratchet has no baseline increase.
